### PR TITLE
migrate validation logic (1st PR)

### PR DESCRIPTION
**Pull request by feilin-flexcompute**
add validator to do checking across multiple attributes in Flow360

### CODE REVIEW

File: flow360/component/flow360_params/flow360_params.py [hunk 1]
```diff
@@ -4,7 +4,6 @@
 # This is a temporary measure until flow360_temp models are merged
 # pylint: disable=too-many-lines
 # pylint: disable=unused-import
-from __future__ import annotations
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/flow360_params.py:7]: This is needed when you refer to type before definition

File: flow360/component/flow360_params/solvers.py [hunk 2]
```diff
@@ -186,17 +186,25 @@ def _get_optional_objects(cls) -> List[str]:
...
+class TurbulenceModelConstantsSST(Flow360BaseModel):
+    """:class:`TurbulenceModelConstantsSST` class"""
+
     C_DES1: Optional[float]
     C_DES2: Optional[float]
     C_d1: Optional[float]
     C_d2: Optional[float]
 
 
+TurbulenceModelConstants = Union[TurbulenceModelConstantsSA, TurbulenceModelConstantsSST]
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/solvers.py:205]: `TurbulenceModelConstants` seems not be used, right?

File: flow360/component/flow360_params/solvers.py [hunk 3]
```diff
@@ -289,6 +297,7 @@ class TurbulenceModelSolverSST(TurbulenceModelSolver):
     """:class:`TurbulenceModelSolverSST` class"""
 
     model_type: Literal["kOmegaSST"] = pd.Field("kOmegaSST", alias="modelType", const=True)
+    model_constants: Optional[TurbulenceModelConstantsSST] = pd.Field(alias="modelConstants")
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/solvers.py:300]: every field present as an allowed type in Flow360Params structure, will be present in webUI available option. Therefore I think the best would be to add defaults to the `TurbulenceModelConstantsSA` and `TurbulenceModelConstantsSST`

File: flow360/component/flow360_params/solvers.py [hunk 4]
```diff
@@ -380,7 +390,7 @@ class TransitionModelSolver(GenericFlowSolverSettings):
         alias="maxForceJacUpdatePhysicalSteps"
     )
     order_of_accuracy: Optional[Literal[1, 2]] = pd.Field(alias="orderOfAccuracy")
-    turbulence_intensity_percent: Optional[PositiveFloat] = pd.Field(
+    turbulence_intensity_percent: Optional[pd.confloat(ge=0.03, le=2.5)] = pd.Field(
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/solvers.py:393]: do we have default here?

File: flow360/component/flow360_params/solvers.py [hunk 5]
```diff
@@ -380,7 +390,7 @@ class TransitionModelSolver(GenericFlowSolverSettings):
...
         alias="turbulenceIntensityPercent"
     )
-    N_crit: Optional[PositiveFloat] = pd.Field(alias="Ncrit")
+    N_crit: Optional[pd.confloat(ge=1, le=11)] = pd.Field(alias="Ncrit")
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/solvers.py:396]: do we have default here?

File: flow360/component/flow360_params/solvers.py (hunk 4 above)
Comment (feilin-flexcompute) [reply flow360/component/flow360_params/solvers.py:393]: No.

File: flow360/component/flow360_params/solvers.py (hunk 5 above)
Comment (feilin-flexcompute) [reply flow360/component/flow360_params/solvers.py:396]: yes

File: flow360/component/flow360_params/solvers.py (hunk 3 above)
Comment (feilin-flexcompute) [reply flow360/component/flow360_params/solvers.py:297]: yes, added.

File: flow360/component/flow360_params/solvers.py (hunk 2 above)
Comment (feilin-flexcompute) [reply flow360/component/flow360_params/solvers.py:205]: it is used in the base class of turbulence model solver, but it is useless, so removed.

File: flow360/component/flow360_params/solvers.py (hunk 5 above)
Comment (feilin-flexcompute) [reply flow360/component/flow360_params/solvers.py]: added

File: flow360/component/flow360_params/solvers.py (hunk 2 above)
Comment (feilin-flexcompute) [reply flow360/component/flow360_params/solvers.py:205]: restored due to model_constants is required for legacy

**Review by maciej-flexcompute — APPROVED** [review]

**Pull request merged**